src/modules/picklist/PickListComponent.js

```diff
@@ -46,6 +51,68 @@ export class PickListContainer extends React.Component {
     this.showMoveNodes = this.showMoveNodes.bind(this);
   }
 
+  componentWillMount() {
+    this.scanner = subscribeToScanner((barCode) => {
+      this.props.clear();
+      const museumId = this.props.appSession.getMuseumId();
+      const collectionId = this.props.appSession.getCollectionId();
+      const token = this.props.appSession.getAccessToken();
+      const isNodeView = this.isTypeNode();
+      if (barCode.uuid) {
+        if (isNodeView) {
+          this.props.findByUUID({uuid: barCode.code, museumId, token})
+            .do((response) => {
+              if (!response) {
+                this.props.emitError({message: I18n.t('musit.errorMainMessages.scanner.noMatchingNode')});
+                return;
+              }
+              const isMoveDialogActive = document.getElementsByClassName('moveDialog').length > 0;
```

Comment:
This feels like a code smell! Is it easy to send in this state so we don't need to interact with the DOM.